### PR TITLE
Make chai-webdriverio compatible with WebdriverIO v5

### PR DESCRIPTION
**Pull request by AnnikaNissen**
This pull request fixes issue https://github.com/marcodejongh/chai-webdriverio/issues/48 and makes `chai-webdriverio` compatible with the new version 5 of WebdriverIO. The changes introduced with WebdriverIO 5 can be found [here](https://github.com/webdriverio/webdriverio/blob/master/CHANGELOG.md#v500-2018-12-20 ).

The pull request introduces the following changes (as also stated in the change log):

**Breaking changes** 
- `.visible()` is renamed to `.displayed()` to be in line with new WebdriverIO wording

**Added feature**
- Compatability with WebdriverIO v5
- TypeScript types are now shipped directly with the `chai-webdriverio` package

**Bug Fixes:**
- `.focus()`, `.text()`, `.value()` exssertions do now wait until [at least one] selected element matches the assertion, not only until [at least one] element with the given selector exists.

### CODE REVIEW

**Comment by mltsy** [comment]
Great update!  I'm not sure about switching to a mirrored versioning scheme, since that would make it impossible to indicate breaking changes in this library before a new version of webdriverio comes out.  My instinct would be to just move it to 1.0.0 and update the Readme with a version compatibility section.  But then, I'm not really maintaining this library anymore so if you have had luck with mirrored version schemes and would like to restrict breaking updates of this library to the release cycle of webdriverio, then feel free to indicate the new versioning scheme concept in the README to clarify for other contributors and keep it at 5.0.0!  Also, could you add to the README that this new version (1.0.0 or 5.0.0, whichever it is) won't support Node versions < `8.x` and update `.travis.yml` to indicate node_js version `8.15.1`?  In fact, if you choose to stick with a mirrored versioning scheme (5.0.0) then I would be tempted to test with `10.x` (since 8.x reaches EOL in December) and indicate that it may work with 8.x indefinitely, but 8.x compatibility going forward is not guaranteed for future updates of major version 5.

**Comment by AnnikaNissen** [comment]
@mltsy 
Thanks for your feedback! 

I'm not sure about the versioning, but I see your point regarding breaking changes in this plugin, so maybe making this a 1.0.0 would be the better choice. I don't have any experiences with versioning - this is my first contribution to open source ever.

I already saw that the travis-build broke because of the npm version. Will definitely take a look at that and update `.travis.yml` and `README`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 # Change Log
 
+## 1.0.0:
+**Breaking changes** 
+- `.visible()` is renamed to `.displayed()` to be in line with new WebdriverIO wording
+
```
Comment (mltsy) [inline CHANGELOG.md:7]: We should add the Node 8.x requirement here under "Breaking Changes" too

**Comment by mltsy** [comment]
First contribution to open source ever! 🎉  Wow, well nice work, this looks a lot better than my first open-source contribution, which I'm pretty sure was like 2 lines, or more likely a documentation update 😉  Ope Source has been missing OUT! ⭐️

**Comment by mltsy** [comment]
@marcodejongh - I think this is ready to merge and release as v1.0.0 if you have a moment sometime to make the release (or perhaps you want to release it as `1.0.0-rc.1` just in case there are issues that require more breaking changes).

**Comment by rpii** [comment]
Thanks for updating this package.

Is this going to be merged?  I have been waiting for this functionality.

**Comment by mltsy** [comment]
@rpii - see https://github.com/marcodejongh/chai-webdriverio/issues/49 (perhaps you'd like to take over releases, or know someone who might be up to it?)

**Comment by rpii** [comment]
Joe,

I am just new to webdriverio and seeing if its suitable for us.  Need this technology bad  to port over many selenium java tests.
I am not comfortable taking over releases at this point.

Rich

**Comment by mltsy** [comment]
No problem.  I don't do node development anymore so I have none of the tools installed.  I don't know how long it will be before Marco might deploy this.  I'd also hate to release a v1.0 indicating it's "production-ready" if there's nobody around to manage releases, so unfortunately you might have to wait for someone who is up for it.

**Comment by marcodejongh** [comment]
@mltsy I thought I gave you access to the npm module?

Let me know if this is ready and I'll merge & publish it

**Comment by marcodejongh** [comment]
Actually looked good to me, published! 🎉

**Comment by mltsy** [comment]
Yay! 😄 (Yeah, sorry, I couldn't remember if you had or not, but when I went to look I realized that I didn't have npm or even Node installed, because I don't use either of them anymore - so I'm kind of out on implementation tasks for this one - just trying to help with communication when I can.  Glad you were able to get to it!)

**Pull request merged**